HospitalSystem.ConsoleUI/Secretary/CheckupUI.cs

```diff
@@ -114,4 +118,138 @@ public void CheckRequests()
+        
+        var referralPatient = patient.MedicalRecord.Referrals[0].Patient;
+        var referralDoctor = patient.MedicalRecord.Referrals[0].Doctor;
+        var referralAnamnesis =  patient.MedicalRecord.AnamnesisHistory;
+        var anamnesis = CreateSentence(referralAnamnesis);
+        var date = EnterDate();
+        var busyAppointments = _hospital.AppointmentService.GetCheckupsByDay(date);
+        busyAppointments.RemoveAll(c => c.Doctor.Id.ToString() != referralDoctor.Id.ToString());
+        var allAppointments = GetAllApointments(date);
+
+        Console.Clear();
+
+        RemoveBusyAppointments(busyAppointments, allAppointments);
+        RemovePastAppointments(allAppointments);
+        ShowFreeAppointments(allAppointments);
+
+        var appointment = EnterAppointment(allAppointments);
+        var dateTime = new DateTime(appointment.Year, appointment.Month, appointment.Day, appointment.Hour, appointment.Minute, appointment.Second);
+
+        Checkup check = new Checkup(dateTime, referralPatient, referralDoctor, anamnesis);
+        _hospital.AppointmentService.UpsertCheckup(check);
+
+        System.Console.Write("Successfully created appointment. Press anything to continue.");
+        ReadSanitizedLine();
+    }
+
+    public string EnterPatientFirstName(){
+        System.Console.Write("Enter patient first name: ");
+        string firstName = ReadSanitizedLine();
+        return firstName;
+    }
+
+    public string EnterPatientLastName(){
+        System.Console.Write("Enter patient last name: ");
+        string lastName = ReadSanitizedLine();
+        return lastName;
+    }
+
+    public void CheckIfPatientExist( List<Patient> patients, string firstName, string lastName){
+        bool success = patients.Any(patient => patient.FirstName == firstName && patient.LastName == lastName);
+        if(!success){
+            throw new InvalidInputException("This patient does not exist");
+        }
+    }
+
+    public string CreateSentence(List<string> referralAnamnesis)
+    {
+        string sentence = "";
+        if(referralAnamnesis.Count != 0)
+        {
+            foreach(var anamnesis in referralAnamnesis)
+            {
+                sentence = sentence +  (char.ToUpper(anamnesis[0]) + anamnesis.Substring(1) + ", ");
+            }
+        }
+        return sentence;
+    }
+
+    public DateTime EnterDate()
+    {
+        Console.Write("Enter date(dd-MM-yyyy): ");
+        string date = ReadSanitizedLine();
+
+        bool success = DateTime.TryParseExact(date, "dd-MM-yyyy", CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime result);
+
+        if (!success) 
+        {
+            throw new InvalidInputException("Wrong date format.");  
+        }
+
+        if (DateTime.Compare(result.Date, DateTime.Now.Date) == -1 )
+        {
+            throw new InvalidInputException("The date is in the past. ");
+        }
```

Comment:
Better to just write < instead of CompareTo, easier to understand

Reply:
One use is so you can pass a comparator to a function that needs it. E.g. you would could bass a lambda like this to sort: (x,y) => -x.CompareTo(y)